Description: Explicitly check for SO_NOSIGPIPE
Origin: vendor
Forwarded: no
Author: Alessandro Ghedini <ghedo@debian.org>
Reviewed-by: Alessandro Ghedini <ghedo@debian.org>
Last-Update: 2012-06-06

diff --git a/src/stream_engine.cpp b/src/stream_engine.cpp
index fa372c9..d9b91cf 100644
--- a/src/stream_engine.cpp
+++ b/src/stream_engine.cpp
@@ -78,7 +78,7 @@ zmq::stream_engine_t::stream_engine_t (fd_t fd_, const options_t &options_) :
 #endif
     }
 
-#if defined ZMQ_HAVE_OSX || defined ZMQ_HAVE_FREEBSD
+#ifdef SO_NOSIGPIPE
     //  Make sure that SIGPIPE signal is not generated when writing to a
     //  connection that was already closed by the peer.
     int set = 1;
